addons/mail/static/src/components/activity/activity.xml

```diff
@@ -47,51 +47,57 @@
                             <i class="fa fa-info-circle" role="img" title="Info"/>
                         </a>
                     </div>
 
                     <t t-if="activityView.areDetailsVisible">
                         <div class="o_Activity_details">
-                            <dl class="dl-horizontal">
-                                <t t-if="activityView.activity.type">
-                                    <dt>Activity type</dt>
-                                    <dd class="o_Activity_type">
+                            <div class="d-table table table-sm mt-2 mb-3">
+                                <div class="d-table-row" t-if="activityView.activity.type">
+                                    <b class="d-table-cell text-right py-1 px-4">Activity type</b>
+                                    <span class="o_Activity_type d-table-cell py-1 pr-4">
                                         <t t-esc="activityView.activity.type.displayName"/>
-                                    </dd>
-                                </t>
-                                <t t-if="activityView.activity.creator">
-                                    <dt>Created</dt>
-                                    <dd class="o_Activity_detailsCreation">
-                                        <t t-esc="activityView.formattedCreateDatetime"/>
-                                        <img class="o_Activity_detailsUserAvatar o_Activity_detailsCreatorAvatar" t-attf-src="/web/image/res.users/{{ activityView.activity.creator.id }}/avatar_128" t-att-title="activityView.activity.creator.nameOrDisplayName" t-att-alt="activityView.activity.creator.nameOrDisplayName"/>
-                                        <span class="o_Activity_detailsCreator">
+                                    </span>
+                                </div>
+                                <div class="d-table-row" t-if="activityView.activity.creator">
+                                    <b class="d-table-cell text-right py-1 px-4">Created</b>
+                                    <span class="o_Activity_detailsCreation d-table-cell py-1 pr-4">
+                                        <t t-esc="activityView.formattedCreateDatetime"/>, by&#160;
+                                        <img class="o_Activity_detailsUserAvatar o_Activity_detailsCreatorAvatar rounded-circle" t-attf-src="/web/image/res.users/{{ activityView.activity.creator.id }}/avatar_128" t-att-title="activityView.activity.creator.nameOrDisplayName" t-att-alt="activityView.activity.creator.nameOrDisplayName"/>
+                                        <b class="o_Activity_detailsCreator">
                                             <t t-esc="activityView.activity.creator.nameOrDisplayName"/>
+                                        </b>
+                                    </span>
+                                </div>
+                                <div class="d-table-row" t-if="activityView.activity.assignee">
```

Comment:
just to make sure: the `t-if` order guideline applies to all of them in the diff of course